### PR TITLE
Write the VFP testing information into Debug file.

### PR DESCRIPTION
**Pull request by qilicun**
These VFP messages should really go into debug files, that's not useful for the user.

```
VFPPROD bhp versus thp not monotonic increasing: 7/4224(0%) elements failed test
VFPPROD bhp versus thp not monotonic increasing: 9/3168(0%) elements failed test
VFPPROD bhp versus thp not monotonic increasing: 1/7296(0%) elements failed test
VFPPROD bhp versus thp not monotonic increasing: 6/15200(0%) elements failed test
VFPPROD bhp versus thp not monotonic increasing: 5/13300(0%) elements failed test
VFPPROD bhp versus thp not monotonic increasing: 11/13300(0%) elements failed test
VFPPROD bhp versus thp not monotonic increasing: 8/13300(0%) elements failed test
VFPPROD bhp versus thp not monotonic increasing: 10/13300(0%) elements failed test
VFPPROD bhp versus thp not monotonic increasing: 15/13680(0%) elements failed test
VFPPROD bhp versus thp not monotonic increasing: 3/15200(0%) elements failed test
VFPPROD bhp versus thp not monotonic increasing: 73/3360(2%) elements failed test
VFPPROD bhp versus thp not monotonic increasing: 335/6804(4%) elements failed test
VFPPROD bhp versus thp not monotonic increasing: 32/6804(0%) elements failed test
VFPPROD bhp versus thp not monotonic increasing: 54/6804(0%) elements failed test
VFPPROD bhp versus thp not monotonic increasing: 282/6912(4%) elements failed test
VFPPROD bhp versus thp not monotonic increasing: 87/6804(1%) elements failed test
VFPPROD bhp versus thp not monotonic increasing: 337/6048(5%) elements failed test

````

### CODE REVIEW

File: opm/parser/eclipse/EclipseState/Tables/VFPProdTable.cpp [hunk 1]
```diff
@@ -386,10 +386,10 @@ void VFPProdTable::check() {
 
     if (num_decreasing > 0) {
         //TODO: Replace with proper log message
```
Comment (atgeirr) [inline opm/parser/eclipse/EclipseState/Tables/VFPProdTable.cpp]: This comment could finally be removed...

**Comment by atgeirr** [comment]
For this message, adding a tag (to eliminate typically > 10 messages) could be considered.

File: opm/parser/eclipse/EclipseState/Tables/VFPProdTable.cpp (hunk 1 above)
Comment (qilicun) [inline opm/parser/eclipse/EclipseState/Tables/VFPProdTable.cpp]: That sounds a final solution! If this should be moved, no tags are needed.

**Comment by atgeirr** [comment]
I just realized the MessageContainer does not support a tag mechanism (I think), please disregard my last message.

File: opm/parser/eclipse/EclipseState/Tables/VFPProdTable.cpp [hunk 2]
```diff
@@ -527,6 +527,17 @@ void VFPProdTable::convertALQToSI(const ALQ_TYPE& type,
...
 
 
+const MessageContainer& VFPProdTable::getMessageContainer() const
+{
+    return m_messages;
+}
+
+
+
+MessageContainer& VFPProdTable::getMessageContainer()
```
Comment (joakim-hove) [inline opm/parser/eclipse/EclipseState/Tables/VFPProdTable.cpp]: Is this mutable reference really needed - if not I would prefer if it were removed.

File: opm/parser/eclipse/EclipseState/Tables/TableManager.cpp [hunk 3]
```diff
@@ -490,6 +490,8 @@ namespace Opm {
 
             VFPProdTable table;
             table.init(keyword, unit_system);
+            auto message = table.getMessageContainer();
+            m_messages.appendMessages(message);
```
Comment (joakim-hove) [inline opm/parser/eclipse/EclipseState/Tables/TableManager.cpp]: Save some characters: `m_messages.appendMessages( table.getMessageContainer() ); `

**Comment by joakim-hove** [comment]
jenkins build this downstreams please

**Comment by joakim-hove** [comment]
OK - good; apart from the veeery small comments I am ready to merge this.

**Comment by qilicun** [comment]
Remove wrong file, please hold.

**Comment by qilicun** [comment]
Ok, this is ready to merge.

**Comment by OPMUSER** [comment]
No this should be in the print file, as it will cause the model to "hunt"
between THP and BHP control. So the user needs be aware of this. That is
the approach used in Eclipse. This is an important warning message.

On Mon, Jun 6, 2016 at 6:30 PM, Liu Ming <notifications@github.com> wrote:

> These VFP messages should really go into debug files, that's not useful
> for the user.
>
> VFPPROD bhp versus thp not monotonic increasing: 7/4224(0%) elements failed test
> VFPPROD bhp versus thp not monotonic increasing: 9/3168(0%) elements failed test
> VFPPROD bhp versus thp not monotonic increasing: 1/7296(0%) elements failed test
> VFPPROD bhp versus thp not monotonic increasing: 6/15200(0%) elements failed test
> VFPPROD bhp versus thp not monotonic increasing: 5/13300(0%) elements failed test
> VFPPROD bhp versus thp not monotonic increasing: 11/13300(0%) elements failed test
> VFPPROD bhp versus thp not monotonic increasing: 8/13300(0%) elements failed test
> VFPPROD bhp versus thp not monotonic increasing: 10/13300(0%) elements failed test
> VFPPROD bhp versus thp not monotonic increasing: 15/13680(0%) elements failed test
> VFPPROD bhp versus thp not monotonic increasing: 3/15200(0%) elements failed test
> VFPPROD bhp versus thp not monotonic increasing: 73/3360(2%) elements failed test
> VFPPROD bhp versus thp not monotonic increasing: 335/6804(4%) elements failed test
> VFPPROD bhp versus thp not monotonic increasing: 32/6804(0%) elements failed test
> VFPPROD bhp versus thp not monotonic increasing: 54/6804(0%) elements failed test
> VFPPROD bhp versus thp not monotonic increasing: 282/6912(4%) elements failed test
> VFPPROD bhp versus thp not monotonic increasing: 87/6804(1%) elements failed test
> VFPPROD bhp versus thp not monotonic increasing: 337/6048(5%) elements failed test
>
>
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/OPM/opm-parser/pull/828
> Commit Summary
>
>    - Write the VFR testing information into Debug file.
>
> File Changes
>
>    - *M* opm/parser/eclipse/EclipseState/Tables/TableManager.cpp
>    <https://github.com/OPM/opm-parser/pull/828/files#diff-0> (2)
>    - *M* opm/parser/eclipse/EclipseState/Tables/VFPProdTable.cpp
>    <https://github.com/OPM/opm-parser/pull/828/files#diff-1> (19)
>    - *M* opm/parser/eclipse/EclipseState/Tables/VFPProdTable.hpp
>    <https://github.com/OPM/opm-parser/pull/828/files#diff-2> (6)
>
> Patch Links:
>
>    - https://github.com/OPM/opm-parser/pull/828.patch
>    - https://github.com/OPM/opm-parser/pull/828.diff
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/OPM/opm-parser/pull/828>, or mute the thread
> <https://github.com/notifications/unsubscribe/ARIIDycV6Df3WPIu4sLKscy8QjW8kAE1ks5qI_bMgaJpZM4IuwvQ>
> .
>



-- 
Regards

David Baxendale
david.baxendale@gmail.com

**Comment by alfbr** [comment]
Thanks for the input! The message as it stands definitely needs to go to debug or be rewritten. We need messages that are informative and clearly explains to the user what is going on.

Maybe you have a suggestion as to how the message could best be formatted?

**Comment by OPMUSER** [comment]
I would suggest something like this:

VFPPROD Table No 1 Monotonic Warning: BHP versus THP not monotonically
increasing:
At Point  (FLOW, THP, WFR, GFR, ALQ) =   16   1   3   2   1  at  BHP =
4.52700E+03
​Warning may cause convergence/hunting between control modes

For reference the Eclipse warning message in the print file is:

 @--WARNING  AT TIME        0.0   DAYS    ( 1-JAN-2004):
 @           PRODUCTION WELL VFP TABLE   8 CONTAINS AT LEAST
ONE
 @           POINT WHERE THE BHP INCREASES WITH DECREASING
THP.
 @           AT POINT (FLOW, THP, WFR, GFR, ALQ) =   16   1   3   2
1
 @           THE BHP =   4.52700E+03   PSIA
.
 @           IF THIS BHP VALUE INDICATES THE FLOW IS CHOKED
OR
 @           EXCEEDS EROSION VELOCITY LIMITS, IT MAY BE
CORRECT
 @           AND THE CHECK MAY BE THRESHOLDED WITH KEYWORD
VFPCHK.
 @           OTHERWISE YOU SHOULD EXAMINE THE TABLE WITH VFPi
AS
 @           IT COULD CAUSE WELL CONVERGENCE OR HUNTING
PROBLEMS.

**Comment by alfbr** [comment]
That looks good, thanks! We will slightly modify it and include line number and path to file where the table is found. We may also want to have short version of the message in the Log-file/stdout.

**Pull request merged**